src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -1,2 +1,1083 @@
 // Reviewing this file for assignment
+package teammates.logic.api;
 
```

Comment:
This class has over 1000 lines of code. It might be more modular and simple to read if the class was boiled down into more subclasses that had distinct functionality! This class could still be the central class that calls on the others, but could definitely be reduced down some.